tests/Context/ApiContextTest.php

```diff
@@ -36,4 +43,36 @@ public function testApiContextSaveRestore()
 
         static::assertEquals($apiContextJson, $apiContextRestored->toJson());
     }
+
+    /**
+     */
+    public function testAutoUpdateBunqContext()
+    {
+        $apiContext = BunqContext::getApiContext();
+
+        $contextJson = json_decode($apiContext->toJson(), true);
+        $expireTime =
+            DateTime::createFromFormat(
+                SessionContext::FORMAT_MICROTIME,
+                $contextJson[ApiContext::FIELD_SESSION_CONTEXT][SessionContext::FIELD_EXPIRY_TIME]
+            );
+        $expireTime->sub(new \DateInterval(self::DATE_TIME_INTERVAL_ONE_YEAR));
+        $contextJson[ApiContext::FIELD_SESSION_CONTEXT][SessionContext::FIELD_EXPIRY_TIME] =
+            $expireTime->format(SessionContext::FORMAT_MICROTIME);
+
+        $expiredApiContext = ApiContext::fromJson(json_encode($contextJson));
+        $validApiContext = clone $expiredApiContext;
+
+        static::assertTrue($validApiContext->ensureSessionActive());
+        static::assertNotEquals($expiredApiContext, $validApiContext);
+
+        BunqContext::updateApiContext(clone $expiredApiContext);
```

Comment:
Why cloning here again @OGKevin ?

Reply:
@OGKevin removed the asserts intentionally? The one on session state can remain right?

Reply:
@sandervdo ☝️

Reply:
Ack